packages/apidom-ns-openapi-2/src/refractor/visitors/SpecificationVisitor.ts

```diff
@@ -1,79 +1,77 @@
-import stampit from 'stampit';
 import { pathSatisfies, path, pick } from 'ramda';
-import { isFunction, isUndefined } from 'ramda-adjunct';
+import { isFunction } from 'ramda-adjunct';
 import { visit, cloneDeep } from '@swagger-api/apidom-core';
 
-import Visitor from './Visitor';
 import { keyMap, getNodeType } from '../../traversal/visitor';
+import Visitor, { VisitorOptions } from './Visitor';
+import FallbackVisitor from './FallbackVisitor';
+import type specification from '../specification';
 
 /**
  * This is a base Type for every visitor that does
  * internal look-ups to retrieve other child visitors.
  */
-const SpecificationVisitor = stampit(Visitor, {
-  props: {
-    specObj: null,
-    passingOptionsNames: ['specObj'],
-  },
-  init({
-    // @ts-ignoreh
-    specObj = this.specObj,
-  }) {
+export interface SpecificationVisitorOptions extends VisitorOptions {
+  readonly specObj: typeof specification;
+}
+
+class SpecificationVisitor extends Visitor {
+  protected readonly specObj!: typeof specification;
```

Comment:
Definitive assignment shouldn't be needed as we're setting it inside the constructor now explicitly. I've issued a PR for existing code: https://github.com/swagger-api/apidom/pull/3736

```suggestion
  protected readonly specObj: typeof specification;
```